panel/widgets/input.py

```diff
@@ -370,6 +372,33 @@ def _process_param_change(self, msg):
         return msg
 
 
+class ArrayInput(LiteralInput):
+
+    max_array_size = param.Number(default=1000, doc="""
+        maximum number of elements in the array to allow full serialization
+        else only the string representation will be passed to the javascript
+        side and edition capabilities through the widget are lost.
+    """)
+
+    _rename = dict(**LiteralInput._rename, **{'max_array_size': None})
+
+
+    def _process_property_change(self, msg):
+        msg = super()._process_property_change(msg)
+        if 'value' in msg and isinstance(msg['value'], list):
+            msg['value'] = np.asarray(msg['value'])
+        return msg
+
+    def _process_param_change(self, msg):
+        if 'value' in msg and msg['value'] is not None:
+            if msg['value'].size <= self.max_array_size:
+                msg['value'] = msg['value'].tolist()
+        msg = super()._process_param_change(msg)
+        if 'value' in msg:
+            msg['value'] = msg['value'].replace(' ', ', ')
```

Comment:
Can you add a comment above this line explaining why it's needed? Will it cause any issues for arrays of strings that might have spaces or commas?

Reply:
Also a bit confused here, shouldn't the `.tolist()` above deal with this since only arrays have a repr that drops the commas?

Reply:
I forgot to drop the replace, I previously used it on the array repr, but with the tolist it is not needed anymore